### DISCUSSION TITLE
Supervised 학습을 위한 Labelling된 데이터를 모을 때 데이터의 이름을 순차적으로 할 필요가 있을까요?

### DISCUSSION THREAD

**Discussion by EcoFriendlyAppleSu** [f-lab-edu/JBLY#99] [General]
💬 학습시킬 데이터의 모집이 중요한 것이지 데이터의 순서는 중요하지 않다고 생가합니다. 따라서, productType_UUID를 사용해 이미지의 이름을 사용하는 것이 어떨까요?

**Comment by Parksojeongjeong** [comment]
uuid를 이용해 파일 이름을 생성한 결과 동일한 상품의 사진이 연속적으로 저장되는 문제가 있어 md5 함수 사용했습니다!

**Reply by EcoFriendlyAppleSu** [reply]
> 매 초 10억개의 uuid를 100년에 걸쳐서 생성할 때 단 하나의 uuid가 중복될 확률은 50%이다. 

> MD5는 128비트 암호화 해시 함수이며 주로 프로그램이나 파일이 원본 그대로인지 확인하는 무결성 검사 등에 사용된다.

위와같이 웹에서 설명을 해주네요.
 상세 상품의 이미지를 긁어오는 부분에서 uuid, md를 사용한 걸로 아는데 어떤 차이가 있길래 연속적으로 저장되는 문제가 해결이된건지 설명해주실 수 있을까요?

**Reply by Parksojeongjeong** [reply]
정확히 하면 이미지를 긁어오는 부분이 아니라 크롤링 한 이미지 url을 파일로 변환하는 과정에서 사용했습니다. 'urllib.request'모듈을 사용해서 크롤링한 이미지 url을 열고 내용을 읽은 후 바이트 객체로 변환했습니다. 이후 md5해시를 계산해서 지정된 폴더로 이미지 파일을 저장했고요. 

바이트 객체로 변환한 후 해시 알고리즘을 이용해서 폴더에 저장해야 하는데 그 과정에서 실수가 있어서 uuid를 사용했을 때 문제가 있었던 걸로 기억합니다. 뿐만 아니라 os error도 같이 발생했었는데 그 과정에서 md5로 변환하는게 더 낫겠다고 판단돼서 수정했습니다!
